Documentation/troubleshooting.rst

```diff
@@ -546,6 +546,82 @@ Opening the ``status``, we can drill down through ``policy.realized.l4``. Do
 your ``ingress`` and ``egress`` rules match what you expect? If not, the
 reference to the errant rules can be found in the ``derived-from-rules`` node.
 
+etcd (kvstore)
+==============
+
+Understanding etcd status
+-------------------------
+
+The etcd status is reported when running ``cilium status``. The following line
+represents the status of etcd:
+
+.. code:: bash
+
+   KVStore:  Ok  etcd: 1/1 connected, lease-ID=29c6732d5d580cb5, lock lease-ID=29c6732d5d580cb7, has-quorum=true: https://192.168.33.11:2379 - 3.4.9 (Leader)
+
+OK:
+  The overall status. Either ``OK`` or ``Failure``.
+
+1/1 connected:
+  Number of total etcd endpoints and how many of them are reachable.
+
+lease-ID:
+  UUID of the lease used for all keys owned by this agent
+
+lock lease-ID:
+  UUID of the lease used for locks acquired by this agent
+
+has-quorum:
+  Status of etcd quorum. Either ``true`` or set to an error.
+
+consecutive-errors:
+  Number of consecutive quorum errors. Only printed if errors are present.
+
+https://192.168.33.11:2379 - 3.4.  9 (Leader):
+  List of all etcd endpoints stating the etcd version and whether the
+  particular endpoint is currently the elected leader. If an etcd endpoint
+  cannot be reached, the error is shown.
+
+Recovery behavior
+-----------------
+
+In the event of an etcd endpoint becoming unhealthy, etcd should automatically
+resolve this by electing a new leader and by failing over to a healthy etcd
+endpoint. As long as quorum is preserved, the etcd cluster will remain
+functional.
+
+In addition, Cilium performs a background check in an interval to determine
+etcd health and potentially take action. The interval depends on the overall
+cluster size. The larger the cluster, the longer the interval:
+
+ * If no etcd endpoints can be reached, Cilium will report failure in ``cilium
+   status``. This will cause the liveness and readiness probe of Kubernetes to
+   fail and Cilium will be restarted.
```

Comment:
We currently have [0] which is fairly outdated as well. We can of course improve it.

We are not introducing significant downtime of the agent but it will result in proxies being restarted. This was already possible before this PR when no etcd endpoint was reachable so a total etcd outage always resulted in agent restarts. We are extending the definition of etcd outage to also including loss of quorum over a long period of time and failure to receive updates for a long period of time.

[0] https://docs.cilium.io/en/v1.8/concepts/failure_behavior/

Reply:
Looks like this is resolved by folding the linked doc into this troubleshooting section.